pkg/server/profiler/cpuprofiler.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package profiler
+
+import (
+	"context"
+	"os"
+	"runtime/pprof"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/server/debug"
+	"github.com/cockroachdb/cockroach/pkg/server/dumpstore"
+	"github.com/cockroachdb/cockroach/pkg/settings"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/errors"
+)
+
+var maxCombinedCPUProfFileSize = settings.RegisterByteSizeSetting(
+	settings.TenantWritable,
+	"server.cpu_profile.total_dump_size_limit",
+	"maximum combined disk size of preserved CPU profiles",
+	128<<20, // 128MiB
+)
+
+var cpuUsageCombined = settings.RegisterIntSetting(
+	settings.TenantWritable,
+	"server.cpu_profile.cpu_usage_combined_threshold",
+	"a threshold beyond which if the combined cpu usage is above, "+
+		"then a cpu profile can be triggered",
```

Comment:
> a threshold (0-100)

You could say that a value above 100 disables this. What about threshold 0? Does it also disable or does it take one every `server.cpu_profile.interval`? Explain all of these.